src/viadot/sources/s3.py

```diff
@@ -310,3 +310,24 @@ def download(self, from_path: str, to_path: str) -> None:
             to_path (str): Path to local file(s) to be stored.
         """
         wr.s3.download(boto3_session=self.session, path=from_path, local_file=to_path)
+
+    def get_page_iterator(
+        self, bucket_name: str, directory_path: str, **kwargs
+    ) -> Iterator[dict[str, Any]]:
+        """Returns an iterator to paginate through the objects in S3 bucket directory.
+
+        This method uses the S3 paginator to list objects under a specified directory
+        path in a given S3 bucket. It can accept additional optional parameters
+        through **kwargs, which will be passed to the paginator.
+
+        Args:
+            bucket_name (str): The name of the S3 bucket.
+            directory_path (str): The directory path (prefix) in the bucket to list
+                objects from.
+            **kwargs: Additional arguments to pass to the paginator (optional).
+
+        Returns:
+            Iterator: An iterator to paginate through the S3 objects.
+        """
+        paginator = wr.s3.get_paginator("list_objects_v2")
```

Comment:
You need to pass boto3 session, same as other methods do -- otherwise, `S3.credentials` is not used - instead, boto3 attempts to read credentials from your machine (~/.aws/credentials`, env vars). Which means this might work on your machine, but not on another.

Reply:
Done